simple_slice_cython.pyx

```diff
@@ -9,7 +9,7 @@ from libc.stdint cimport int64_t
 import sys
 
 cdef extern from "Python.h":
-    int PyIndex_Check(object obj)
+    Py_ssize_t PyNumber_AsSsize_t(object obj, object exc) except? -1
```

Comment:
Minor, trivial: can in principle simplify to use `from cpython.number cimport PyNumber_AsSsize_t`, also for other functions.